core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -55,6 +55,13 @@ oppia.factory('UrlService', ['$window', function($window) {
       }
       throw Error('Invalid topic id url');
     },
+    getTopicNameFromLearnerUrl: function() {
+      var pathname = this.getPathname();
+      if (pathname.match(/\/(story|topic)/g)) {
+        return pathname.split('/')[2];
+      }
+      throw Error('Invalid topic name');
```

Comment:
Invalid URL for topic.

(It's the URL that's invalid, not the topic name.)

Reply:
Done.